elfo-logger/src/actor.rs

```diff
@@ -120,61 +124,79 @@ impl Logger {
         }
     }
 
-    pub(super) fn format_event<T: theme::Theme>(&mut self, event: PreparedEvent) {
-        let out = &mut self.buffer;
+    pub(super) fn format_event<T: theme::Theme, F: LineFactory>(&mut self, event: PreparedEvent) {
         let config = self.ctx.config();
-
-        let payload = self
-            .shared
-            .pool
-            .get(event.payload_id)
-            .expect("unknown string");
-        self.shared.pool.clear(event.payload_id);
-
-        // <timestamp> <level> [<trace_id>] <object> - <message>\t<fields>
-
-        T::Timestamp::fmt(out, &event.timestamp);
-        out.push(' ');
-        T::Level::fmt(out, event.metadata.level());
-        out.push_str(" [");
-        T::TraceId::fmt(out, &event.trace_id);
-        out.push_str("] ");
-        T::ActorMeta::fmt(out, &event.object);
-        out.push_str(" - ");
-        T::Payload::fmt(out, &payload);
-
-        // Add ancestors' fields.
-        let mut span_id = event.span_id;
-        while let Some(data) = span_id
-            .as_ref()
-            .and_then(|span_id| self.shared.spans.get(span_id))
-        {
-            span_id.clone_from(&data.parent_id);
+        let try_slow_path = {
+            let mut line = F::create_line(&mut self.buffer);
```

Comment:
I would prefer use different function for this instead of recursion as more obvious (ad, hence, less error-prone) approach:
```rust
fn format_event<T: theme::Theme>(&mut self, event: PreparedEvent) {
    let fits = self.do_format_event<T, FailOnUnfit>(&event);

    if !fits {
        self.do_format_event<T, TruncateOnUnfit>(&event);
    }

    // Cleanup
    self.shared.pool.clear(event.payload_id);
}

fn do_format_event<T: theme::Theme, L: LineLimiter>(&mut self, event: &PreparedEvent) -> bool {
    // ...
    line.try_approve()
}
```

Or even move rerun login to `LineBuffer` (that's require tracking spans of parts inside the underlying buffer), but it requires more It takes more effort.

Anyway, I can accept even the current approach if there is no desire to make code cleaner

Reply:
Done by [0719951](https://github.com/elfo-rs/elfo/pull/128/commits/07199518dc9f07d23c83b9438f2099c974519acc)